gitbutler-ui/src/lib/vbranches/tooltip.ts

```diff
@@ -0,0 +1,23 @@
+import type { Commit } from './types';
+
+export function getLockText(commitId: string[], commits: Commit[]): string;
+export function getLockText(commitId: string, commits: Commit[]): string;
+export function getLockText(commitId: any, commits: Commit[]): string {
+	if (!commitId || commits === undefined) return 'Depends on a committed change';
```

Comment:
Is the typescript overloading syntax something we want to go for? With it not actually having any polymorphism to back it up, found it to be a more cumbersome alternative to using a union type:
```ts
export function getLockText(commitId: string[] | string | any, commits: Commit[]): string {
```


Style aside, I am wondering why we need the `any` type and whether `undefined` would suffice.

Reply:
Thanks!